patches/api/0382-More-Teleport-API.patch

```diff
@@ -0,0 +1,279 @@
++
++    /**
++     * Causes the player to look towards the given position.
++     *
++     * @param x x coordinate
++     * @param y y coordinate
++     * @param z z coordinate
++     * @param playerAnchor What part of the player should face the given position
++     */
++    void lookAt(double x, double y, double z, @NotNull io.papermc.paper.entity.LookAnchor playerAnchor);
++
++    /**
++     * Causes the player to look towards the given location.
++     *
++     * @param location Location to look at
++     * @param playerAnchor What part of player should face the location
++     */
++    default void lookAt(@NotNull Location location, @NotNull io.papermc.paper.entity.LookAnchor playerAnchor) {
++        lookAt(location.getX(), location.getY(), location.getZ(), playerAnchor);
++    }
++
++    /**
++     * Causes the player to look towards the given entity.
++     *
++     * @param entity Entity to look at
++     * @param playerAnchor What part of the player should face the entity
++     * @param entityAnchor What part of the entity the player should face
++     */
++    void lookAt(@NotNull org.bukkit.entity.Entity entity, @NotNull io.papermc.paper.entity.LookAnchor playerAnchor,  @NotNull io.papermc.paper.entity.LookAnchor entityAnchor);
++    // Paper end - Teleport API
++
+     @NotNull
+     @Override
+     Spigot spigot();
+diff --git a/src/main/java/org/bukkit/event/player/PlayerTeleportEvent.java b/src/main/java/org/bukkit/event/player/PlayerTeleportEvent.java
+index 553d7740489fe729166c8ca8ef8c7834db3663ad..f1780588807689727dda1691c82e9aa4d297035a 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerTeleportEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerTeleportEvent.java
+@@ -13,8 +13,14 @@ public class PlayerTeleportEvent extends PlayerMoveEvent {
+     private static final HandlerList handlers = new HandlerList();
+     private TeleportCause cause = TeleportCause.UNKNOWN;
+ 
++    // Paper start - Teleport API
++    private boolean dismounted = true;
++    private final java.util.Set<io.papermc.paper.entity.RelativeTeleportFlag> teleportFlagSet;
++    // Paper end
++
+     public PlayerTeleportEvent(@NotNull final Player player, @NotNull final Location from, @Nullable final Location to) {
+         super(player, from, to);
++        teleportFlagSet = java.util.Collections.emptySet(); // Paper - Teleport API
+     }
+ 
+     public PlayerTeleportEvent(@NotNull final Player player, @NotNull final Location from, @Nullable final Location to, @NotNull final TeleportCause cause) {
+@@ -23,6 +29,16 @@ public class PlayerTeleportEvent extends PlayerMoveEvent {
+         this.cause = cause;
+     }
+ 
++    // Paper start - Teleport API
++    public PlayerTeleportEvent(@NotNull final Player player, @NotNull final Location from, @Nullable final Location to, @NotNull final TeleportCause cause, boolean dismounted, @NotNull java.util.Set<io.papermc.paper.entity.RelativeTeleportFlag> teleportFlagSet) {
```

Comment:
![image](https://user-images.githubusercontent.com/23108066/167048081-945fe37d-ec14-422d-ac8a-c7f241f4ea91.png)

Reply:
I believe that's because the full class name is used. From my testing, if you write `io.papermc.paper.entity.@NotNull RelativeTeleportFlag` inside the generic, it should work.